sdk/cosmos/azure-cosmos/azure/cosmos/_global_endpoint_manager.py

```diff
@@ -89,29 +89,29 @@ def force_refresh(self, database_account):
         self.refresh_endpoint_list(database_account)
 
     def refresh_endpoint_list(self, database_account, **kwargs):
-        with self.refresh_lock:
-            # if refresh is not needed or refresh is already taking place, return
-            if not self.refresh_needed:
-                return
-            try:
-                self._refresh_endpoint_list_private(database_account, **kwargs)
-            except Exception as e:
-                raise e
+        if self.refresh_needed:
+            with self.refresh_lock:
+                # if refresh is not needed or refresh is already taking place, return
+                if not self.refresh_needed:
+                    return
+                try:
+                    self._refresh_endpoint_list_private(database_account, **kwargs)
+                except Exception as e:
+                    raise e
 
     def _refresh_endpoint_list_private(self, database_account=None, **kwargs):
+        self.refresh_needed = False
```

Comment:
previously, this flag will only be set if SDK ever get a successful response. After this change, what would happen for the following scenario? 

Request 1 -> refresh account failed -> the request will fail? 
Request 2 -> since the refresh_needed = false due to request 1, what would happen here?

Reply:
the only place that the function could fail in would be calling _GetDatabaseAccount and receiving an error, which would fail the request regardless since we surface the error so there's no problems there